runtime/wasmer_test.go

```diff
@@ -0,0 +1,532 @@
+		return 0, err
+	}
+	defer out.Close()
+
+	resp, err := http.Get("https://github.com/w3f/polkadot-re-tests/blob/master/polkadot-runtime/polkadot_runtime.compact.wasm?raw=true")
+	if err != nil {
+		return 0, err
+	}
+	defer resp.Body.Close()
+
+	n, err = io.Copy(out, resp.Body)
+	return n, err
+}
+
+// Exists reports whether the named file or directory exists.
+func Exists(name string) bool {
+	if _, err := os.Stat(name); err != nil {
+		if os.IsNotExist(err) {
+			return false
+		}
+	}
+	return true
+}
+
+func newRuntime(t *testing.T) (*Runtime, error) {
+	_, err := getRuntimeBlob()
+	if err != nil {
+		t.Fatalf("Fail: could not get polkadot runtime")
+	}
+
+	fp, err := filepath.Abs(POLKADOT_RUNTIME_FP)
+	if err != nil {
+		t.Fatal("could not create filepath")
+	}
+
+	tt := &trie.Trie{}
+
+	r, err := NewRuntime(fp, tt)
+	if err != nil {
+		t.Fatal(err)
+	} else if r == nil {
+		t.Fatal("did not create new VM")
+	}
+
+	return r, err
+}
+
+func TestExecVersion(t *testing.T) {
+	expected := &Version{
+		Spec_name:         []byte("polkadot"),
+		Impl_name:         []byte("parity-polkadot"),
+		Authoring_version: 1,
+		Spec_version:      1000,
+		Impl_version:      0,
+	}
+
+	r, err := newRuntime(t)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ret, err := r.Exec("Core_version", 1, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Log(ret)
+
+	res, err := decodeToInterface(ret, &Version{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	version := res.(*Version)
+	t.Logf("Spec_name: %s\n", version.Spec_name)
+	t.Logf("Impl_name: %s\n", version.Impl_name)
+	t.Logf("Authoring_version: %d\n", version.Authoring_version)
+	t.Logf("Spec_version: %d\n", version.Spec_version)
+	t.Logf("Impl_version: %d\n", version.Impl_version)
+
+	if !reflect.DeepEqual(version, expected) {
+		t.Errorf("Fail: got %v expected %v\n", version, expected)
+	}
+}
+
+const TESTS_FP string = "./test_wasm/target/wasm32-unknown-unknown/release/test_wasm.wasm"
+const TESTS_FP_2 string = "./test_wasm.wasm"
+
+// getTestBlob checks if the test wasm file exists and if not, it fetches it from github
+func getTestBlob() (n int64, err error) {
+	if Exists(TESTS_FP) || Exists(TESTS_FP_2) {
+		return 0, nil
+	}
+
+	out, err := os.Create(TESTS_FP_2)
+	if err != nil {
+		return 0, err
+	}
+	defer out.Close()
+
+	resp, err := http.Get("https://github.com/ChainSafe/gossamer-test-wasm/raw/master/target/wasm32-unknown-unknown/release/test_wasm.wasm")
+	if err != nil {
+		return 0, err
+	}
+	defer resp.Body.Close()
+
+	n, err = io.Copy(out, resp.Body)
+	return n, err
+}
+
+func newTestRuntime() (*Runtime, error) {
+	_, err := getTestBlob()
+	if err != nil {
+		return nil, err
+	}
+
+	t := &trie.Trie{}
+	fp, err := filepath.Abs(TESTS_FP_2)
+	if err != nil {
+		return nil, err
+	}
+	r, err := NewRuntime(fp, t)
+	if err != nil {
+		fp, err := filepath.Abs(TESTS_FP)
+		if err != nil {
+			return nil, err
+		}
+		return NewRuntime(fp, t)
+	}
+
+	return r, nil
+}
+
+func TestExt_get_storage_into(t *testing.T) {
```

Comment:
I start to have trouble following the code here. This could benefit hugely from some consistent and more declarative variable names (eg. `keyPos` instead of `keyData`). Some comments to describe setup and how we validate a pass would make this a lot more readable.

Reply:
the variables are named to directly match the variable names of the ext functions, which I got from the spec, but yeah I'll add comments

Reply:
I have added comments to the tests. let me know if anything is still unclear